content/kubeone/master/cheat_sheets/production_recommendations/_index.en.md

```diff
@@ -108,3 +108,70 @@ resource "azurerm_lb" "lb" {
 
 [azure-lb-skus]: https://docs.microsoft.com/en-us/azure/load-balancer/skus
 [azure-lb]: https://github.com/kubermatic/kubeone/blob/c3121b7482f910327ef15b187735e79de0bc9572/examples/terraform/azure/main.tf#L143-L157
+
+## Hetzner
+
+### Kubernetes API Server Endpoint
+
+{{% notice warning %}}
+**This can be applied only when creating the cluster for the first time.
+Doing this on an existing cluster will effectively break the cluster!**
+{{% /notice %}}
+
+By default, the provided Terraform configs for Hetzner use Load Balancer's
+public IP for the Kubernetes API server endpoint. This ensures that the
+Kubernetes API is reachable via the Internet (e.g. from you local machine using
+`kubectl`), however, this causes the traffic between Kubernetes Services to go
+via the public interface (i.e. via the Internet).
+
+On Hetzner, there's no option to disable the public IP/interface. The only way
+to enforce the traffic between Service to be routed via the private interface
+is to use the private interface for the Kubernetes API server endpoint.
+This can be done by providing the Load Balancer's private IP (instead of the
+public IP) as the Kubernetes API server endpoint.
+
+Doing this will effectively disable accessing the Kubernetes API via the
+Internet, but that can be solved by creating an SSH tunnel to one of the
+instances. KubeOne provides `kubeone proxy` command which can do that for you.
+
+If you're using our Terraform configs, you only need to change value in the
+[`kubeone_api` section of `output.tf`][hz-output-tf] to
+`hcloud_load_balancer.load_balancer.network_ip`.
```

Comment:
hcloud_load_balancer_network.load_balancer.0.ip

Reference: https://registry.terraform.io/providers/hetznercloud/hcloud/latest/docs/resources/load_balancer_network#ip

The load_balancer resource can't provide the private ip.

Reply:
And this value should be put in main.tf:20

Reply:
Also, kubeone must add the loadbalancer's public ip to the certificates that get generated.

Reply:
UPDATE: managed to get this configured, still pods talk using the public interface.

Reply:
Hetzner CCM must be installed before the network CNI. With Network CNI deployed before, the CCM will not be able to control it and it will always use primary interface.

Reply:
This PR is on hold because the instructions are not working at all. We'll revisit this in the future as part of https://github.com/kubermatic/kubeone/issues/1388.